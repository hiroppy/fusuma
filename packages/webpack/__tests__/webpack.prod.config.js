const { isAbsolute } = require('path');
const rewire = require('rewire');

describe('webpack.prod', () => {
  test('should match settings', () => {
    const prod = rewire('../src/webpack.prod.config');

    process.env.NDOE_ENV = 'production';
    prod.__set__('__dirname', 'stabDir');

    const res = prod();

    // can not stub using rewire
    res.module.rules.forEach((rule) => {
      if (rule.test.toString() === '/\\.css$/') {
        rule.use.forEach((r, i) => {
          if (typeof r === 'string' && isAbsolute(r)) rule.use[i] = 'stub/';
        });
      }
    });

    // res.plugins.forEach((p) => {
    //   console.log(p.constructor.name);
    //   if (p.constructor.name === 'MiniCssExtractPlugin') {
    //   }
    // });

    expect(res).toMatchSnapshot();

    process.env.NDOE_ENV = 'test';
  });
});
