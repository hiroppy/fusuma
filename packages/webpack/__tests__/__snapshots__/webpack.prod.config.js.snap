// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack.prod should match settings 1`] = `
Object {
  "mode": "production",
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\\\\\.css\\$/,
        "use": Array [
          "stub/",
          Object {
            "loader": "css-loader",
            "options": Object {
              "importLoaders": 2,
              "modules": false,
            },
          },
          Object {
            "loader": "postcss-loader",
            "options": Object {
              "plugins": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
            },
          },
        ],
      },
    ],
  },
  "optimization": Object {
    "minimizer": Array [
      TerserPlugin {
        "options": Object {
          "cache": false,
          "cacheKeys": [Function],
          "chunkFilter": [Function],
          "exclude": undefined,
          "extractComments": false,
          "include": undefined,
          "minify": undefined,
          "parallel": true,
          "sourceMap": false,
          "terserOptions": Object {
            "output": Object {
              "comments": /\\^\\\\\\*\\*!\\|@preserve\\|@license\\|@cc_on/i,
            },
          },
          "test": /\\\\\\.m\\?js\\(\\\\\\?\\.\\*\\)\\?\\$/i,
          "warningsFilter": [Function],
        },
      },
      OptimizeCssAssetsWebpackPlugin {
        "deleteAssetsMap": Object {},
        "options": Object {
          "assetNameRegExp": /\\\\\\.css\\(\\\\\\?\\.\\*\\)\\?\\$/i,
          "assetProcessors": Array [
            Object {
              "phase": "compilation.optimize-chunk-assets",
              "processor": [Function],
              "regExp": /\\\\\\.css\\(\\\\\\?\\.\\*\\)\\?\\$/i,
            },
          ],
          "canPrint": undefined,
          "cssProcessor": [Function],
          "cssProcessorOptions": Object {
            "discardComments": Object {
              "removeAll": true,
            },
            "safe": true,
          },
          "cssProcessorPluginOptions": Object {},
        },
        "phaseAssetProcessors": Object {
          "compilation.optimize-assets": Array [],
          "compilation.optimize-chunk-assets": Array [
            Object {
              "phase": "compilation.optimize-chunk-assets",
              "processor": [Function],
              "regExp": /\\\\\\.css\\(\\\\\\?\\.\\*\\)\\?\\$/i,
            },
          ],
          "emit": Array [],
        },
        "pluginDescriptor": Object {
          "name": "OptimizeCssAssetsWebpackPlugin",
        },
      },
    ],
    "runtimeChunk": "single",
    "splitChunks": Object {
      "cacheGroups": Object {
        "vendor": Object {
          "chunks": "initial",
          "enforce": true,
          "name": "vendor",
          "test": /node_modules/,
        },
      },
    },
  },
  "output": Object {
    "chunkFilename": "[name].[id].[chunkhash].bundle.js",
    "filename": "[name].[chunkhash].bundle.js",
  },
  "plugins": Array [
    MiniCssExtractPlugin {
      "options": Object {
        "chunkFilename": "[name].[id].[chunkhash].css",
        "filename": "[name].[chunkhash].css",
        "ignoreOrder": false,
        "moduleFilename": [Function],
      },
    },
  ],
}
`;
